### DISCUSSION TITLE
Support new_value_mode "add" for dict options

### DISCUSSION THREAD

**Discussion by me21** [zauberzeug/nicegui#2462] [Ideas / Feature Requests]
Hi,

I tried to use dict options in ui.select along with the ability to add new options by the user.
During the control initialization, I got an error that new_value_mode "add" is not supported for dict options.
I think we could be able to supply an addition callback which returns new value id, something like:

```py
def id_generator(starting_from: int):
    def my_generator(new_value) -> int:
        # in theory, we could use "new_value" which user has entered to derive its id from it
        id = starting_from
        while True:
            yield id
            id += 1
    return my_generator 

ui.select({1: 'Alice', 2: 'Bob'}, 
             label='User', 
             new_value_mode='add', 
             new_value_id_generator=id_generator(3))  # will assign new ids to user entered values starting from 3 onwards - just for example - can be written shorter as itertools.count(start=3, step=1)
```

As another option, it could be a real callback function, or an iterator.

What do you think?

**Comment by me21** [comment]
My bad, reading through the code I found that 'add-unique' mode is supported. But feel free to add this if you feel it would make the select control better.

**Comment by me21** [comment]
Made a pull request: https://github.com/zauberzeug/nicegui/pull/2467

**Comment by falkoschindler** [comment]
This feature just got released in [version 1.4.18](https://github.com/zauberzeug/nicegui/releases/tag/v1.4.18).
